src/utils/tools/redisTools.ts

```diff
@@ -0,0 +1,57 @@
+import { IconInterface } from 'components/icons'
+import { RedisClient } from 'redis'
+
+// This array should have the keys that can have values other than string
+// Category might be a string or an array, that's why it's added.
+// Note: types should be taken from the IconInterface
+const nonStringFields = ['tags', 'category', 'hasOutlined']
+
+// For Redis Hash: The keys and values must be strings, that's why we need to stringify other types.
+const serialize = (icon: IconInterface) => {
+  for (const key of Object.keys(icon)) {
+    if (nonStringFields.includes(key)) {
+      icon[key] = JSON.stringify(icon[key])
+    }
+  }
+}
+
+const deserialize = (icon: IconInterface) => {
+  for (const key of Object.keys(icon)) {
+    if (nonStringFields.includes(key)) {
+      icon[key] = JSON.parse(icon[key])
+    }
+  }
+}
+
+/**
+    **********************************************
+    Promise style Redis Functions, to avoid using callbacks
+    **********************************************
+*/
+
+function asyncHset (redisClient: RedisClient, name: string, key: string, value: string) {
```

Comment:
you can use the built-in Node.js `util.promisify` for this without having to write your own Promise.

FYI: Almost every function/lib can be converted to promise while are following Node.js error first, callback pattern =)

Reply:
I've tried using util.promisify, it didn't work, this's strange, I think there's a nested calls to other internal redis  methods which caused the error, I will give it another try, maybe I did something wrong.

Reply:
I figured it, I just needed to bind the `redis client`, since redis using `this` keyword internally. 
Anyway, Promisify doesn't handle multiple arguments smoothly, we would need to hack it too, the alternative I've used for `svgToImg` was Pify package, the total time for the update process increased 700 ms after using it, I think it's not really needed, what do you think? @en3sis

Reply:
Don't worry, it was more about you not having to write your own wrapper but do it if you have to =)